scripts/get_raw_eumetsat_data.py

```diff
@@ -0,0 +1,354 @@
+############
+# Pull raw satellite data from EUMetSat
+#
+# 2021-09-28
+# Jacob Bieker
+#
+############
+
+import fsspec
+import yaml
+from satip import eumetsat
+import pandas as pd
+import os
+import math
+from satpy import Scene
+
+from typing import Optional, List, Tuple
+
+import re
+from datetime import datetime, timedelta
+
+from nowcasting_dataset.consts import SAT_VARIABLE_NAMES
+import click
+
+# SatPy gives a UserWarning about Proj, which isn't actually used here, so can be
+# safely ignored
+import warnings
+
+warnings.simplefilter("ignore", UserWarning)
+
+
+NATIVE_FILESIZE_MB = 102.210123
+CLOUD_FILESIZE_MB = 3.445185
+RSS_ID = "EO:EUM:DAT:MSG:MSG15-RSS"
+CLOUD_ID = "EO:EUM:DAT:MSG:RSS-CLM"
+
+format_dt_str = lambda dt: pd.to_datetime(dt).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+def validate_date(ctx, param, value):
+    try:
+        return format_dt_str(value)
+    except ValueError:
+        raise click.BadParameter("Date must be in format accepted by pd.to_datetime()")
+
+
+@click.command()
+@click.option(
+    "--download_directory",
+    "--dir",
+    default="/run/media/jacob/data/",
+    help="Where to download the data to. Also where the script searches for previously downloaded data.",
+)
+@click.option(
+    "--start_date",
+    "--start",
+    default="2021-01-01",
+    prompt="Starting date to download data, in format accepted by pd.to_datetime()",
+    callback=validate_date,
+)
+@click.option(
+    "--end_date",
+    "--end",
+    default="2021-01-10",
+    prompt="Ending date to download data, in format accepted by pd.to_datetime()",
+    callback=validate_date,
+)
+@click.option(
+    "--backfill",
+    "-b",
+    default=False,
+    prompt="Whether to download any missing data from the start date of the data on disk to the end date",
+    is_flag=True,
+)
+@click.option(
+    "--user_key",
+    "--key",
+    default=None,
+    help="The User Key for EUMETSAT access",
```

Comment:
Maybe add a sentence to the `help` to say something like `Alternatively, the user key can be set using an auth file.` ?

Reply:
Done!